reana_commons/validation/utils.py

```diff
@@ -32,18 +57,47 @@ def validate_reana_yaml(reana_yaml: Dict) -> None:
     """
     try:
         with open(reana_yaml_schema_file_path, "r") as f:
+            # Create validator from REANA specification schema
             reana_yaml_schema = json.loads(f.read())
-            validate(reana_yaml, reana_yaml_schema)
+            validator_class = validator_for(reana_yaml_schema)
+            validator_class.check_schema(reana_yaml_schema)
+            validator = validator_class(reana_yaml_schema)
+
+            # Collect all validation errors
+            errors = [e for e in validator.iter_errors(reana_yaml)]
+
+            non_critical_validators = ["additionalProperties"]
+            # Depending on whether a validator is critical or not,
+            # separate errors into 'critical' and 'warnings'
+            critical_errors = []
+            # The warning dictionary has as keys the properties that are not
+            # respected, and as values, a list of strings that invalidates the property
+            # or describe the error
+            warnings = {}
+            for e in errors:
+                if e.validator in non_critical_validators:
+                    warnings.setdefault(e.validator, []).extend(
```

Comment:
What do you think about "translating" the validator name from camelcase to snakecase (`additionalProperties` -> `additional_properties`) so that we are consistent with the rest of the values returned by the API?